### ISSUE TITLE
Drawer component only styled properly when using bottom-opening direction

### ISSUE THREAD

**Issue by camball** [huntabyte/shadcn-svelte#1273]
## TL;DR

The `direction` prop from the [vaul-svelte](https://github.com/huntabyte/vaul-svelte) component is honoured in shadcn-svelte's `Drawer` component, but has some hard-coded styling making any direction other than a bottom-opening drawer unusable.

## Motivation

[vaul-svelte](https://github.com/huntabyte/vaul-svelte)'s `Drawer.Root` [takes](https://github.com/huntabyte/vaul-svelte?tab=readme-ov-file#root) a `direction` prop:

> `direction`: Direction of the drawer. Can be `top`, `bottom`, `left`, or `right`. Defaults to `bottom`.

When attempting to pass a `direction` to shadcn-svelte's `Drawer.Root`

```svelte
<Drawer.Root direction="top">
  ...
</Drawer.Root>
```

, the prop is honoured, yet there are fixed default classes that mandate the direction coming from the bottom, making the style appear incorrectly

https://github.com/huntabyte/shadcn-svelte/blob/cd3902ec7bf1bd1b4b6a3320058bd16cf5e2ca1a/sites/docs/src/lib/registry/default/ui/drawer/drawer-content.svelte#L16

Expected behaviour here is to honour the prop and style the component correctly, as shown in the [vaul-svelte website examples](https://www.vaul-svelte.com/examples).

## Solution

If we dynamically apply the proper directional tailwind classes, based on the passed direction, the shadcn-svelte `Drawer` wrapper will work correctly for any passed direction, out of the box.

### Drawer Handle Considerations

One thing that I feel would look wrong is the ["drawer handle"](https://github.com/huntabyte/shadcn-svelte/blob/cd3902ec7bf1bd1b4b6a3320058bd16cf5e2ca1a/sites/docs/src/lib/registry/default/ui/drawer/drawer-content.svelte#L21) being present on left- and right-opening drawers. IMO, from a design perspective, it should only be present on top and bottom drawers.

### Proposed PR Implementation

Will need to modify shadcn-svelte's [drawer-content.svelte](https://github.com/huntabyte/shadcn-svelte/blob/main/sites/docs/src/lib/registry/default/ui/drawer/drawer-content.svelte):

- To use one of `{bottom, top, left, right}-0` instead of a fixed `bottom-0` class, as well as one of `rounded-{b, t, l, r}-[10px]` instead of the fixed `rounded-t-[10px]`.
- Remove the "drawer handle" for left and right, only keeping it for top and bottom. Provide an option to remove the drawer handle entirely as well. Currently, there's an `mt-24` class applied (to account for the size of the drawer handle), which would also need to be changed dynamically, based on drawer direction, but since I believe the handle should only be present on the top and bottom drawers, it should only every either be `mt-24` or `mb-24` (only applied when the drawer is present).
- See if we can reuse the [`DrawerDirection`](https://github.com/huntabyte/vaul-svelte/blob/c061dda9123df0681a80dffde97ce7b2eef5cf90/src/lib/vaul/index.ts#L3) type straight from vaul-svelte in the shadcn component.

I'm more than happy to own this issue and open a PR with this proposed implementation if this issue is accepted.

### Reproduction

Working Reproduction in StackBlitz: [Shadcn-Svelte Drawer Direction Styling Bug](https://stackblitz.com/edit/shadcn-svelte-drawer-direction-styling-bug?file=src%2Froutes%2F%2Bpage.svelte)

---

```svelte
<script lang="ts">
  import { Button } from '$lib/components/ui/button';
  import * as Drawer from '$lib/components/ui/drawer';
</script>

<!-- or substitute `direction` with 'left' or 'right' to observe the wrong styling -->
<Drawer.Root direction="top">
  <Drawer.Trigger>
    <Button>Open Drawer</Button>
  </Drawer.Trigger>
  <Drawer.Content>
    <Drawer.Close class="m-10">Close Drawer</Drawer.Close>
  </Drawer.Content>
</Drawer.Root>
```

### System Info

```bash
System:
  OS: macOS 14.6.1
  CPU: (12) arm64 Apple M3 Pro
  Memory: 68.31 MB / 36.00 GB
  Shell: 5.9 - /bin/zsh
Binaries:
  Node: 20.14.0 - ~/.nvm/versions/node/v20.14.0/bin/node
  npm: 10.8.2 - ~/.nvm/versions/node/v20.14.0/bin/npm
Browsers:
  Chrome: 128.0.6613.138
  Safari: 17.6
```

### Severity

annoyance

**Comment by yuriko192** [comment]
hey, not a maintainer here, but i just encountered the same issue as you. my solution is to use drawer solely for bottom direction, while left-right direction i try and use the [Sheet component](https://www.shadcn-svelte.com/docs/components/sheet), hope this might help you too

**Comment by cburwell** [comment]
> hey, not a maintainer here, but i just encountered the same issue as you. my solution is to use drawer solely for bottom direction, while other direction i try and use the [Sheet component](https://www.shadcn-svelte.com/docs/components/sheet), hope this might help you too

Thanks for this! I was having the same issue but Sheet is the exact desired effect for me. I agree that drawer should probably be able to handle directions, but I wasn't even able to get it to override the bottom direction.

**Comment by ieedan** [comment]
@AdrianGonz97 can confirm this is fixed on `@next`
